pkg/secrets-store/provider_client_test.go

```diff
@@ -334,3 +334,42 @@ func TestPluginClientBuilderErrorInvalid(t *testing.T) {
 		t.Errorf("Get(%s) = %v, want %v", "bad/provider/name", err, ErrInvalidProvider)
 	}
 }
+
+func TestVersion(t *testing.T) {
```

Comment:
What do you think about a test case for `EnableHealthCheck` itself? It would be nice to at least have something that covers the `p.lock.RLock()` through `p.lock.RUnlock()`

Reply:
Done!